hc/api/views.py

```diff
@@ -81,8 +82,14 @@ def _update(check, spec):
     if "channels" in spec:
         if spec["channels"] == "*":
             check.assign_all_channels()
-        elif spec["channels"] == "":
+        else:
             check.channel_set.clear()
+            if spec["channels"] is not None and spec["channels"] != "":
+                channels = []
+                for raw_channel in spec["channels"].split(","):
+                    channel = get_object_or_404(Channel, code=UUID(raw_channel))
```

Comment:
If any of the channel codes is invalid, would be good to return 400 instead of 404.

Reply:
Sure. Changed to respond with 400.